From 892571cebada3c257c04e4a5d3fc6f9a3a83e8be Mon Sep 17 00:00:00 2001
From: Stefano Babic <stefano.babic@swupdate.org>
Date: Thu, 3 Jul 2025 14:46:26 +0200
Subject: [PATCH 09/11] settings: cosmetic, group the public functions

Signed-off-by: Stefano Babic <stefano.babic@swupdate.org>

Upstream-Status: Submitted [https://groups.google.com/g/swupdate/c/7f3Q2PAgQPY]
---
 corelib/swupdate_settings.c | 24 ++++++++++++++----------
 1 file changed, 14 insertions(+), 10 deletions(-)

diff --git a/corelib/swupdate_settings.c b/corelib/swupdate_settings.c
index e23c9f29..681efb93 100644
--- a/corelib/swupdate_settings.c
+++ b/corelib/swupdate_settings.c
@@ -71,6 +71,20 @@ static int read_settings_file(config_t *cfg, const char *filename)
 	return ret;
 }
 
+static int get_run_as(void *elem, void *data)
+{
+	struct run_as *pid = (struct run_as *)data;
+
+	GET_FIELD_INT(LIBCFG_PARSER, elem, "userid", (int *)&pid->userid);
+	GET_FIELD_INT(LIBCFG_PARSER, elem, "groupid", (int *)&pid->groupid);
+
+	return 0;
+}
+
+/*
+ * Public interface for the Settings
+ * These are the functions exported to the other modules
+ */
 int read_module_settings(swupdate_cfg_handle *handle, const char *module, settings_callback fcn, void *data)
 {
 	config_setting_t *elem;
@@ -91,16 +105,6 @@ int read_module_settings(swupdate_cfg_handle *handle, const char *module, settin
 	return 0;
 }
 
-static int get_run_as(void *elem, void *data)
-{
-	struct run_as *pid = (struct run_as *)data;
-
-	GET_FIELD_INT(LIBCFG_PARSER, elem, "userid", (int *)&pid->userid);
-	GET_FIELD_INT(LIBCFG_PARSER, elem, "groupid", (int *)&pid->groupid);
-
-	return 0;
-}
-
 int read_settings_user_id(swupdate_cfg_handle *handle, const char *module, uid_t *userid, gid_t *groupid)
 {
 	struct run_as ids;
-- 
2.44.3

