From 97303a45b704a198c7b3915c0025fbc8a8f303c7 Mon Sep 17 00:00:00 2001
From: Michael Glembotzki <Michael.Glembotzki@iris-sensing.com>
Date: Tue, 19 Apr 2022 11:32:41 +0200
Subject: [PATCH] imx8mp-evk: Add cmd for booting signed fitImage

Signed-off-by: Michael Glembotzki <Michael.Glembotzki@iris-sensing.com>
---
 include/configs/imx8mp_evk.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/configs/imx8mp_evk.h b/include/configs/imx8mp_evk.h
index 94035d74c9..08fbfd4dcd 100644
--- a/include/configs/imx8mp_evk.h
+++ b/include/configs/imx8mp_evk.h
@@ -140,6 +140,9 @@
 		"source\0" \
 	"loadimage=fatload mmc ${mmcdev}#${mmcpart} ${loadaddr} ${image}\0" \
 	"loadfdt=fatload mmc ${mmcdev}#${mmcpart} ${fdt_addr} ${fdt_file}\0" \
+	"fit_img=fitImage.signed\0" \
+	"fit_addr=0x48000000\0" \
+	"fitboot=fatload mmc ${mmcdev}#${mmcpart} ${fit_addr} ${fit_img}; run mmcargs; bootm ${fit_addr}\0" \
 	"mmcboot=echo Booting from mmc ...; " \
 		"run mmcargs; " \
 		"if test ${boot_fit} = yes || test ${boot_fit} = try; then " \
-- 
2.39.0

