From 2a2a487a75f9719061112e3c5543377359d90722 Mon Sep 17 00:00:00 2001
From: Michael Glembotzki <Michael.Glembotzki@iris-sensing.com>
Date: Thu, 14 Apr 2022 22:45:04 +0200
Subject: [PATCH] Add HAB image authentication for FIT Images

Signed-off-by: Michael Glembotzki <Michael.Glembotzki@iris-sensing.com>
---
 cmd/bootm.c | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/cmd/bootm.c b/cmd/bootm.c
index 0c5585b123..e65664eeed 100644
--- a/cmd/bootm.c
+++ b/cmd/bootm.c
@@ -163,8 +163,14 @@ int do_bootm(struct cmd_tbl *cmdtp, int flag, int argc, char *const argv[])
 	}
 
 #else
+	ulong img_addr;
+	if(argc == 0) {
+		img_addr = image_load_addr;
+	} else {
+		img_addr = simple_strtoul(argv[0], NULL, 16);
+	}
 
-	switch (genimg_get_format((const void *)image_load_addr)) {
+	switch (genimg_get_format((void*) img_addr)) {
 #if defined(CONFIG_LEGACY_IMAGE_FORMAT)
 	case IMAGE_FORMAT_LEGACY:
 		if (authenticate_image(image_load_addr,
@@ -178,6 +184,15 @@ int do_bootm(struct cmd_tbl *cmdtp, int flag, int argc, char *const argv[])
 	case IMAGE_FORMAT_ANDROID:
 		/* Do this authentication in boota command */
 		break;
+#endif
+#ifdef CONFIG_FIT
+	case IMAGE_FORMAT_FIT:
+		if (authenticate_image(img_addr,
+			fit_get_size((void *)img_addr)) != 0) {
+			printf("Authenticate FIT image Fail, Please check\n");
+			return 1;
+		}
+		break;
 #endif
 	default:
 		printf("Not valid image format for Authentication, Please check\n");
-- 
2.39.0

