From 8c77aa3d6f3941f41630708b4b0e4618bfde30b0 Mon Sep 17 00:00:00 2001
From: Michael Glembotzki <Michael.Glembotzki@iris-sensing.com>
Date: Tue, 19 Apr 2022 10:45:11 +0200
Subject: [PATCH 15/20] imx8mp-irma6r2: Add cmd for booting signed fitImage

Signed-off-by: Michael Glembotzki <Michael.Glembotzki@iris-sensing.com>
---
 include/configs/imx8mp_irma6r2.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/configs/imx8mp_irma6r2.h b/include/configs/imx8mp_irma6r2.h
index e53d3de64d..551dac6162 100644
--- a/include/configs/imx8mp_irma6r2.h
+++ b/include/configs/imx8mp_irma6r2.h
@@ -123,6 +123,9 @@
 		"source\0" \
 	"loadimage=fatload mmc ${mmcdev}#${mmcpart} ${loadaddr} ${image}\0" \
 	"loadfdt=fatload mmc ${mmcdev}#${mmcpart} ${fdt_addr} ${fdt_file}\0" \
+	"fit_img=fitImage.signed\0" \
+	"fit_addr=0x48000000\0" \
+	"fitboot=fatload mmc ${mmcdev}#${mmcpart} ${fit_addr} ${fit_img}; run mmcargs; bootm ${fit_addr}\0" \
 	"mmcboot=echo Booting from mmc ...; " \
 		"run mmcargs; " \
 		"if test ${boot_fit} = yes || test ${boot_fit} = try; then " \
-- 
2.34.1

