From 409d2dd67d2374b4874cd10831c5982003528425 Mon Sep 17 00:00:00 2001
From: Michael Glembotzki <Michael.Glembotzki@iris-sensing.com>
Date: Thu, 14 Apr 2022 22:43:07 +0200
Subject: [PATCH 16/20] imx8mp-irma6r2: Increase CONFIG_SYS_BOOTM_LEN to 32MB

Signed-off-by: Michael Glembotzki <Michael.Glembotzki@iris-sensing.com>
---
 include/configs/imx8mp_irma6r2.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/configs/imx8mp_irma6r2.h b/include/configs/imx8mp_irma6r2.h
index 551dac6162..6ad7c11d7f 100644
--- a/include/configs/imx8mp_irma6r2.h
+++ b/include/configs/imx8mp_irma6r2.h
@@ -175,6 +175,7 @@
 #define CONFIG_LOADADDR			0x40480000
 
 #define CONFIG_SYS_LOAD_ADDR		CONFIG_LOADADDR
+#define CONFIG_SYS_BOOTM_LEN		0x2000000 /* 32MB */
 
 #define CONFIG_SYS_INIT_RAM_ADDR	0x40000000
 #define CONFIG_SYS_INIT_RAM_SIZE	0x80000
-- 
2.34.1

