From 3197eec01bf6cfbfbf5f6377368b9203b3277890 Mon Sep 17 00:00:00 2001
From: Michael Glembotzki <Michael.Glembotzki@iris-sensing.com>
Date: Tue, 19 Apr 2022 15:04:37 +0200
Subject: [PATCH] imx8mp-irma6r2: Change bootcmd to fitboot

Signed-off-by: Michael Glembotzki <Michael.Glembotzki@iris-sensing.com>
---
 include/configs/imx8mp_irma6r2.h | 13 ++-----------
 1 file changed, 2 insertions(+), 11 deletions(-)

diff --git a/include/configs/imx8mp_irma6r2.h b/include/configs/imx8mp_irma6r2.h
index 6ad7c11d7f..b1254487c2 100644
--- a/include/configs/imx8mp_irma6r2.h
+++ b/include/configs/imx8mp_irma6r2.h
@@ -158,17 +158,8 @@
 			"fi; " \
 		"fi;\0"
 
-#define CONFIG_BOOTCOMMAND \
-	   "mmc dev ${mmcdev}; if mmc rescan; then " \
-		   "if run loadbootscript; then " \
-			   "run bootscript; " \
-		   "else " \
-			   "if run loadimage; then " \
-				   "run mmcboot; " \
-			   "else run netboot; " \
-			   "fi; " \
-		   "fi; " \
-	   "fi;"
+#define CONFIG_BOOTCOMMAND "run fitboot;"
+
 #endif
 
 /* Link Definitions */
-- 
2.34.1

