From 68c227c676c136cbc2d53052ec7bd9e458b44046 Mon Sep 17 00:00:00 2001
From: Michael Glembotzki <Michael.Glembotzki@iris-sensing.com>
Date: Wed, 1 Mar 2023 20:06:56 +0100
Subject: [PATCH] imx8mp-irma6r2: Set lpj=32000 and remove double entry mmcargs

Signed-off-by: Michael Glembotzki <Michael.Glembotzki@iris-sensing.com>
---
 include/configs/imx8mp_irma6r2.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/include/configs/imx8mp_irma6r2.h b/include/configs/imx8mp_irma6r2.h
index d109533770..82a7b34ce3 100644
--- a/include/configs/imx8mp_irma6r2.h
+++ b/include/configs/imx8mp_irma6r2.h
@@ -144,8 +144,7 @@
 	"mmcdev="__stringify(CONFIG_SYS_MMC_ENV_DEV)"\0" \
 	"mmcroot=" CONFIG_MMCROOT " rootwait rw\0" \
 	"mmcautodetect=yes\0" \
-	"mmcargs=setenv bootargs ${jh_clk} ${mcore_clk} console=${console} root=${mmcroot}\0 " \
-	"mmcargs=setenv bootargs ${jh_clk} ${mcore_clk} console=${console} root=${mmcroot} ${mmcpart}\0 " \
+	"mmcargs=setenv bootargs ${jh_clk} ${mcore_clk} console=${console} root=${mmcroot} lpj=32000 ${mmcpart}\0 " \
 	"loadbootscript=fatload mmc ${mmcdev}#${mmcpart} ${loadaddr} ${script};\0" \
 	"bootscript=echo Running bootscript from mmc ...; " \
 		"source\0" \
-- 
2.39.2

