From d8b5d60aa77808a83986a8d825e58a4f2db9d46f Mon Sep 17 00:00:00 2001
From: Michael Glembotzki <Michael.Glembotzki@iris-sensing.com>
Date: Tue, 19 Apr 2022 15:03:59 +0200
Subject: [PATCH] imx8mp-evk: Change bootcmd to fitboot

Signed-off-by: Michael Glembotzki <Michael.Glembotzki@iris-sensing.com>
---
 include/configs/imx8mp_evk.h | 12 +-----------
 1 file changed, 1 insertion(+), 11 deletions(-)

diff --git a/include/configs/imx8mp_evk.h b/include/configs/imx8mp_evk.h
index 79729d79730..8dc7aee9866 100644
--- a/include/configs/imx8mp_evk.h
+++ b/include/configs/imx8mp_evk.h
@@ -144,17 +144,7 @@
 				"echo WARN: Cannot load the DT; " \
 			"fi; " \
 		"fi;\0" \
-	"bsp_bootcmd=echo Running BSP bootcmd ...; " \
-		"mmc dev ${mmcdev}; if mmc rescan; then " \
-		   "if run loadbootscript; then " \
-			   "run bootscript; " \
-		   "else " \
-			   "if run loadimage; then " \
-				   "run mmcboot; " \
-			   "else run netboot; " \
-			   "fi; " \
-		   "fi; " \
-	   "fi;"
+	"bsp_bootcmd=run fitboot;"
 #endif
 
 /* Link Definitions */
-- 
2.44.1

