From 147e2f2ee0aa46d5634948c27f681b67a987a5e8 Mon Sep 17 00:00:00 2001
From: Jasper Orschulko <jasper@fancydomain.eu>
Date: Tue, 23 Feb 2021 17:13:16 +0100
Subject: [PATCH 20/20] add localversion-iris

Signed-off-by: Jasper Orschulko <jasper@fancydomain.eu>
---
 localversion-iris | 1 +
 1 file changed, 1 insertion(+)
 create mode 100644 localversion-iris

diff --git a/localversion-iris b/localversion-iris
new file mode 100644
index 00000000..7fd7513c
--- /dev/null
+++ b/localversion-iris
@@ -0,0 +1 @@
+-gen6-1.1.5
-- 
2.26.3

