From 5f516528d0c4eb524c04e53cd53cfe9c701652ab Mon Sep 17 00:00:00 2001
From: Erik Schumacher <erik.schumacher@iris-sensing.com>
Date: Mon, 29 Jan 2024 14:49:21 +0100
Subject: [PATCH 2/2] Add iris specific checks to wpa-supplicant.sh

---
 ./wpa-supplicant.sh          | 27 +++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/wpa-supplicant.sh b/wpa-supplicant.sh
index 75df8d709e..f15bf7e2ba 100644
--- a/wpa-supplicant.sh
+++ b/wpa-supplicant.sh
@@ -9,6 +9,33 @@ WPA_SUP_OPTIONS="-B -P $WPA_SUP_PIDFILE -i $IFACE -s"
 
 VERBOSITY=0
 
+IRMA_CONFIG="/mnt/iris/counter/config_customer.json"
+EAPTLS_DIR="/mnt/iris/eaptls"
+EAPTLS_KEY="sensor.key"
+EAPTLS_CRT="sensor.crt"
+EAPTLS_CA="sensor.ca"
+
+if [ ! -f "$IRMA_CONFIG" ]; then
+       if [ "$VERBOSITY" = "1" ]; then
+               echo "customer configuration file is missing: $IRMA_CONFIG"
+       fi
+       exit 0
+fi
+
+val=$(jq ".sets.IRMA6_Customer.parameters[\"pa.communication.enable_eaptls\"].value" "$IRMA_CONFIG")
+if [ "$val" != "true" ]; then
+       if [ "$VERBOSITY" = "1" ]; then
+               echo "\"pa.communication.enableeaptls\" flag is disabled or invalid in configuration file"
+       fi
+       exit 0
+fi
+
+if [ ! -d "$EAPTLS_DIR" ] || [ ! -f "$EAPTLS_DIR/$EAPTLS_KEY" ] || [ ! -f "$EAPTLS_DIR/$EAPTLS_CRT" ] || [ ! -f "$EAPTLS_DIR/$EAPTLS_CA" ]; then
+       if [ "$VERBOSITY" = "1" ]; then
+               echo "EAP-TLS is enabled but the key material is missing from $EAPTLS_DIR"
+       fi
+       exit 0
+fi
 
 if [ -s "$IF_WPA_CONF" ]; then
 	WPA_SUP_CONF="-c $IF_WPA_CONF"
-- 
2.43.0

