From f4410c7ccef837a3aede76365e59a1ec05718343 Mon Sep 17 00:00:00 2001
From: Erik Schumacher <erik.schumacher@iris-sensing.com>
Date: Thu, 8 Aug 2024 16:26:54 +0200
Subject: [PATCH] Add flash_fitimage for imx93

---
 iMX93/soc.mak | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/iMX93/soc.mak b/iMX93/soc.mak
index d94d115..02194d8 100644
--- a/iMX93/soc.mak
+++ b/iMX93/soc.mak
@@ -40,6 +40,9 @@ CAPSULE_GUID = bc550d86-da26-4b70-ac05-2a448eda6f21
 FCB_LOAD_ADDR ?= $(ATF_LOAD_ADDR)
 MCU_IMG = m33_image.bin
 
+FITIMAGE ?= irma-fitimage.itb
+FITIMAGE_ADDR ?= $(KERNEL_ADDR)
+
 TEE ?= tee.bin
 TEE_LOAD_ADDR ?= 0x96000000
 MCU_XIP_ADDR ?= 0x28032000 # Point entry of m33 in flexspi0 nor flash
@@ -303,6 +306,9 @@ flash_sentinel: $(MKIMG) ahabfw.bin
 flash_kernel: $(MKIMG) Image $(KERNEL_DTB)
 	./$(MKIMG) -soc IMX9 -c -ap Image a55 $(KERNEL_ADDR) --data $(KERNEL_DTB) a55 $(KERNEL_DTB_ADDR) -out flash.bin
 
+flash_fitimage: $(MKIMG) $(FITIMAGE)
+	./$(MKIMG) -soc IMX9 -c -ap $(FITIMAGE) a55 $(FITIMAGE_ADDR) -out flash_os.bin
+
 flash_bootaux_cntr: $(MKIMG) $(MCU_IMG)
 	./$(MKIMG) -soc IMX9 -c -m33 $(MCU_IMG) 0 $(MCU_TCM_ADDR) $(MCU_TCM_ADDR_ACORE_VIEW) -out flash.bin
 
-- 
2.46.0

