From 91e4c1822ea8e05b69428727c6200f32108479fa Mon Sep 17 00:00:00 2001
From: Michael Glembotzki <Michael.Glembotzki@iris-sensing.com>
Date: Thu, 26 Sep 2024 13:01:51 +0200
Subject: [PATCH] imx93-matrixup.dts: Remove mipi dsi stuff

---
 arch/arm/dts/imx93-matrixup.dts | 56 ---------------------------------
 1 file changed, 56 deletions(-)

diff --git a/arch/arm/dts/imx93-matrixup.dts b/arch/arm/dts/imx93-matrixup.dts
index dd5d0f152b1..82380b8e004 100644
--- a/arch/arm/dts/imx93-matrixup.dts
+++ b/arch/arm/dts/imx93-matrixup.dts
@@ -71,30 +71,6 @@
 		gpio = <&pcal6524 14 GPIO_ACTIVE_HIGH>;
 		enable-active-high;
 	};
-
-	dsi_host: dsi-host {
-		compatible = "synopsys,dw-mipi-dsi";
-		status = "okay";
-	};
-
-	rm67199_panel {
-		compatible = "raydium,rm67199";
-		reset-gpio = <&adp5585gpio 6 GPIO_ACTIVE_LOW>;
-		dsi-lanes = <4>;
-		video-mode = <2>;	/* 0: burst mode
-					 * 1: non-burst mode with sync event
-					 * 2: non-burst mode with sync pulse
-					 */
-		width-mm = <68>;
-		height-mm = <121>;
-		status = "okay";
-
-		port {
-			panel_in: endpoint {
-				remote-endpoint = <&dsi_out>;
-			};
-		};
-	};
 };
 
 &adc1 {
@@ -106,24 +82,6 @@
 	status = "okay";
 };
 
-&dsi {
-	status = "okay";
-
-	ports {
-		port@1 {
-			dsi_to_adv7535: endpoint {
-				remote-endpoint = <&adv7535_to_dsi>;
-			};
-		};
-
-		port@2 {
-			dsi_out: endpoint {
-				remote-endpoint = <&panel_in>;
-			};
-		};
-	};
-};
-
 &mu1 {
 	status = "okay";
 };
@@ -166,20 +124,6 @@
 	pinctrl-1 = <&pinctrl_lpi2c1>;
 	status = "okay";
 
-	adv7535: hdmi@3d {
-		compatible = "adi,adv7535";
-		reg = <0x3d>;
-		adi,addr-cec = <0x3c>;
-		adi,dsi-lanes = <4>;
-		status = "okay";
-
-		port {
-			adv7535_to_dsi: endpoint {
-				remote-endpoint = <&dsi_to_adv7535>;
-			};
-		};
-	};
-
 	pmic@25 {
 		compatible = "nxp,pca9451a";
 		reg = <0x25>;
-- 
2.46.0

