From f76221af1ea1f236f5a99be97e041b64fd62cac6 Mon Sep 17 00:00:00 2001
From: Erik Schumacher <erik.schumacher@iris-sensing.com>
Date: Fri, 27 Sep 2024 07:15:43 +0000
Subject: [PATCH] imx93-matrixup.dts: Remove rtc0 alias

---
 arch/arm/dts/imx93-matrixup.dts | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/arch/arm/dts/imx93-matrixup.dts b/arch/arm/dts/imx93-matrixup.dts
index 79aa3533e86..4541fe21487 100644
--- a/arch/arm/dts/imx93-matrixup.dts
+++ b/arch/arm/dts/imx93-matrixup.dts
@@ -12,10 +12,6 @@
 	model = "NXP i.MX93 MATRIX UP board";
 	compatible = "fsl,imx93-matrixup", "fsl,imx93";
 
-	aliases {
-		rtc0 = &pcf2131;
-	 };
-
 	chosen {
 		bootargs = "console=ttyLP0,115200";
 		stdout-path = &lpuart1;
-- 
2.46.0

