From 0c37176fa8fb7a6f00c09f1b5c98b7b1a136b9bb Mon Sep 17 00:00:00 2001
From: Erik Schumacher <erik.schumacher@iris-sensing.com>
Date: Fri, 11 Oct 2024 13:06:13 +0000
Subject: [PATCH] WIP - use lower speed RAM configuration

---
 board/freescale/imx93_matrixup/spl.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/board/freescale/imx93_matrixup/spl.c b/board/freescale/imx93_matrixup/spl.c
index 8d5017069a8..b234d57b4f3 100644
--- a/board/freescale/imx93_matrixup/spl.c
+++ b/board/freescale/imx93_matrixup/spl.c
@@ -75,6 +75,7 @@ void spl_dram_init(void)
 	if (is_voltage_mode(VOLT_LOW_DRIVE))
 		ptiming = &dram_timing_1866mts;
 #endif
+	ptiming = &dram_timing_1866mts;
 
 	printf("DDR: %uMTS\n", ptiming->fsp_msg[0].drate);
 	ddr_init(ptiming);
-- 
2.46.0

