From 3903e730e172d79963ccf56133b3abd62238a453 Mon Sep 17 00:00:00 2001
From: Erik Schumacher <erik.schumacher@iris-sensing.com>
Date: Thu, 24 Mar 2022 10:47:45 +0100
Subject: [PATCH] Add imx8mp irma6r2 SOC, based on imx8mp with DDR4 firmware

---
 iMX8M/soc.mak | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/iMX8M/soc.mak b/iMX8M/soc.mak
index b7b3986..4e23da8 100644
--- a/iMX8M/soc.mak
+++ b/iMX8M/soc.mak
@@ -62,6 +62,19 @@ QSPI_HEADER = ../scripts/fspi_header
 QSPI_PACKER = ../scripts/fspi_packer.sh
 VERSION = v2
 LPDDR_FW_VERSION = _201904
+else ifeq ($(SOC),iMX8MPI6R2)
+PLAT = imx8mp
+HDMI = no
+SPL_LOAD_ADDR = 0x920000
+SPL_FSPI_LOAD_ADDR = 0x920000
+TEE_LOAD_ADDR =  0x56000000
+ATF_LOAD_ADDR = 0x00970000
+VAL_BOARD = val
+#define the F(Q)SPI header file
+QSPI_HEADER = ../scripts/fspi_header
+QSPI_PACKER = ../scripts/fspi_packer.sh
+VERSION = v2
+DDR_FW_VERSION = _201810
 else
 PLAT = imx8mq
 HDMI = yes
@@ -171,7 +184,7 @@ u-boot-ddr4.itb: $(dtbs_ddr4)
 
 dtbs_ddr4_evk = evkddr4.dtb
 $(dtbs_ddr4_evk):
-	./$(DTB_PREPROC) $(PLAT)-ddr4-evk.dtb $(dtbs_ddr4_evk)
+	./$(DTB_PREPROC) $(PLAT)-irma6r2.dtb $(dtbs_ddr4_evk)
 
 u-boot-ddr4-evk.itb: $(dtbs_ddr4_evk)
 	./$(PAD_IMAGE) tee.bin
-- 
2.34.1

