From 842dc270eed8d7123887320776a77e7c410e0191 Mon Sep 17 00:00:00 2001
From: Michael Glembotzki <Michael.Glembotzki@iris-sensing.com>
Date: Tue, 12 Nov 2024 12:14:33 +0100
Subject: [PATCH] imx93-matrixup: Make u-boot-env eth1addr writable

Signed-off-by: Michael Glembotzki <Michael.Glembotzki@iris-sensing.com>
---
 include/configs/imx93_matrixup.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/configs/imx93_matrixup.h b/include/configs/imx93_matrixup.h
index b4f7fb71572..070894caf83 100644
--- a/include/configs/imx93_matrixup.h
+++ b/include/configs/imx93_matrixup.h
@@ -54,7 +54,8 @@
 	"upgrade_available:dw," \
 	"bootcount:dw," \
 	"ustate:dw," \
-	"firmware:dw"
+	"firmware:dw," \
+	"eth1addr:mw"
 
 /* Initial environment variables */
 #define CFG_EXTRA_ENV_SETTINGS		\
-- 
2.46.1

