From 273fd86adb3b2e5e60ea0da40077410db119dc4a Mon Sep 17 00:00:00 2001
From: Ian Dannapel <ian.dannapel@iris-sensing.com>
Date: Thu, 6 Apr 2023 10:16:44 +0200
Subject: [PATCH] epc660: reload fw after hw reset

Signed-off-by: Ian Dannapel <ian.dannapel@iris-sensing.com>
Signed-off-by: Jasper Orschulko <jasper@fancydomain.eu>
---
 drivers/media/i2c/epc660.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/media/i2c/epc660.c b/drivers/media/i2c/epc660.c
index a8112cb7595a..a45bc9364f5d 100644
--- a/drivers/media/i2c/epc660.c
+++ b/drivers/media/i2c/epc660.c
@@ -388,8 +388,8 @@ static long epc660_ioctl(struct v4l2_subdev *sd, unsigned int cmd, void *arg)
 		}
 		break;
 	case EPC_660_IOCTL_CMD_RESET:
-		epc660_reset_hw(sd, 1);
-		epc660_reset_hw(sd, 0);
+		/* reset and reload fw*/
+		epc660_load_fw(sd);
 		break;
 	default:
 		return -ENOIOCTLCMD;
@@ -434,19 +434,19 @@ static int epc660_load_fw(struct v4l2_subdev *sd)
 		return ret;
 	}
 
-	dev_info(&client->dev, "initialization ");
+	dev_info(&client->dev, "programming init sequence");
 	ret = epc660_send_i2c_sequence(client, epc660_init_sequence);
 	if (ret < 0) {
 		return ret;
 	};
-	dev_info(&client->dev, "initialization done.\n");
-	dev_info(&client->dev, "sequencer programming");
+
+	dev_info(&client->dev, "programming firmware sequencer");
 	ret = epc660_send_i2c_sequence(client,
 				       epc660_003_Seq_Prog_8MHz_Default_8);
 	if (ret < 0) {
 		return ret;
 	};
-	dev_info(&client->dev, "sequencer programming done.\n");
+	dev_info(&client->dev, "firmware sequencer loaded succesfully\n");
 
 	return 0;
 }
-- 
2.46.0

