From fda9b1d518ecd14cb7e04d26fa07e69a2daebd84 Mon Sep 17 00:00:00 2001
From: Erik Schumacher <erik.schumacher@iris-sensing.com>
Date: Mon, 24 Jun 2024 15:09:29 +0000
Subject: [PATCH] epc660.c: Add link frequency v4l2_ctrl

---
 drivers/media/i2c/epc660.c | 17 +++++++++++++++--
 1 file changed, 15 insertions(+), 2 deletions(-)

diff --git a/drivers/media/i2c/epc660.c b/drivers/media/i2c/epc660.c
index 3f0449c5ea08..1bb12f71961d 100644
--- a/drivers/media/i2c/epc660.c
+++ b/drivers/media/i2c/epc660.c
@@ -62,6 +62,7 @@ struct epc660 {
 	struct v4l2_subdev subdev;
 	struct v4l2_ctrl_handler hdl;
 	struct v4l2_ctrl *pixel_rate;
+	struct v4l2_ctrl *link_freq;
 	struct v4l2_ctrl *hblank;
 	struct v4l2_rect rect; /* Sensor window */
 	const struct epc660_datafmt *fmt;
@@ -582,6 +583,13 @@ static struct v4l2_subdev_ops epc660_subdev_ops = {
 	.pad = &epc660_subdev_pad_ops,
 };
 
+#define DEFAULT_PIXEL_RATE 24000000
+
+/* supported link frequencies */
+static const s64 link_freq_menu_items[] = {
+	DEFAULT_PIXEL_RATE,
+};
+
 static int epc660_probe(struct i2c_client *client)
 {
 	struct epc660 *epc660;
@@ -606,14 +614,19 @@ static int epc660_probe(struct i2c_client *client)
 	v4l2_i2c_subdev_init(&epc660->subdev, client, &epc660_subdev_ops);
 	v4l2_ctrl_handler_init(&epc660->hdl, 6);
 
-	pixel_rate_max = 24000000;
-	pixel_rate_min = 24000000;
+	pixel_rate_max = DEFAULT_PIXEL_RATE;
+	pixel_rate_min = DEFAULT_PIXEL_RATE;
 	/* By default, PIXEL_RATE is read only */
 	epc660->pixel_rate =
 		v4l2_ctrl_new_std(&epc660->hdl, &epc660_ctrl_ops,
 				  V4L2_CID_PIXEL_RATE, pixel_rate_min,
 				  pixel_rate_max, 1, pixel_rate_max);
 
+	epc660->link_freq =
+		v4l2_ctrl_new_int_menu(&epc660->hdl, &epc660_ctrl_ops,
+				V4L2_CID_LINK_FREQ, ARRAY_SIZE(link_freq_menu_items) - 1, 0,
+				link_freq_menu_items);
+
 	/*
 	hblank = inactive periods of hsync
 	hblank = (2 * t_CONV - t_HSYNC) * f_PCLK
-- 
2.46.0

